src/Centreon/Infrastructure/ServiceConfiguration/ServiceConfigurationRepositoryRDB.php

```diff
@@ -0,0 +1,143 @@
+     */
+    private $sqlRequestTranslator;
+
+    public function __construct(DatabaseConnection $db, SqlRequestParametersTranslator $sqlRequestTranslator)
+    {
+        $this->db = $db;
+        $this->sqlRequestTranslator = $sqlRequestTranslator;
+        $this->sqlRequestTranslator
+            ->getRequestParameters()
+            ->setConcordanceStrictMode(RequestParameters::CONCORDANCE_MODE_STRICT);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findService(int $serviceId): ?Service
+    {
+        try {
+            $request = $this->translateDbName(
+                'SELECT service_id AS id, service_template_model_stm_id AS template_id, display_name AS name,
+                service_description AS description, service_locked AS is_locked, service_register AS is_registred,
+                service_activate AS is_activated
+                FROM `:db`.service
+                WHERE service_id = :service_id'
+            );
+            $statement = $this->db->prepare($request);
+            $statement->bindValue(':service_id', $serviceId, \PDO::PARAM_INT);
+            $statement->execute();
+            if (($record = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
+                return EntityCreator::createEntityByArray(
+                    Service::class,
+                    $record
+                );
+            }
+            return null;
+        } catch (\Throwable $ex) {
+            throw new RepositoryException('Error while searching for the service', 0, $ex);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findOnDemandServiceMacros(int $serviceId): array
+    {
+        try {
+            $request = $this->translateDbName('
+                SELECT svc_macro_id AS id, svc_macro_name AS name, svc_macro_value AS `value`, macro_order AS `order`,
+                is_password, description
+                FROM `:db`.on_demand_macro_service
+                WHERE svc_svc_id = :service_id
+            ');
+            $statement = $this->db->prepare($request);
+            $statement->bindValue(':service_id', $serviceId, \PDO::PARAM_INT);
+            $statement->execute();
+
+            $serviceMacros = [];
+            while (($record = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
+                $serviceMacros[] = EntityCreator::createEntityByArray(
+                    ServiceMacro::class,
+                    $record
+                );
+            }
+            return $serviceMacros;
+        } catch (\Throwable $ex) {
+            throw new RepositoryException('Error while searching for the on-demand service macros', 0, $ex);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findCommandLine(int $serviceId): ?string
+    {
+        try {
+            $request = $this->translateDbName(
+                'WITH RECURSIVE inherite AS (
+                SELECT service_id, service_template_model_stm_id, command_command_id 
```

Comment:
shouldn't you cast to avoid a "service_template_model_stm_id" = NULL and truncated results in the recursion ?

Reply:
Recursion is performed as long as the service_template_model_stm_id is not null